storage/mempool.go

```diff
@@ -185,6 +185,45 @@ func (s *Storage) GetPendingUserState(stateID uint32) (*models.UserState, error)
 	}, nil
 }
 
+// does not return the full list of states, only the pending c2ts
+func (s *Storage) GetPendingUserStates(pubkey *models.PublicKey) ([]models.UserState, error) {
+	result := make([]models.UserState, 0)
+
+	maxUint32 := ^uint32(0)
+	pubKeyID := maxUint32
+
+	accounts, err := s.AccountTree.Leaves(pubkey)
+	if err == nil && len(accounts) > 0 {
+		// TODO: is this lookup really the right thing to do?
+		pubKeyID = accounts[0].PubKeyID
+	}
```

Comment:
Is it possible that there are multiple pubKeyIDs for a given pubkey?

Reply:
nvm, we just need a valid one if it exists.

Reply:
I'm not quite sure what the "right" thing to do here is, in the normal case there should only be one matching `pubkeyID`. When the transaction is applied we will use the first key: https://github.com/worldcoin/hubble-commander/blob/e5f2708efb623f6292a585bcbd29eab6d1bfaed8/commander/applier/apply_create2transfer.go#L75

so maybe this should reproduce that behavior